src/Crypto/RNCryptor/Types.hs

```diff
@@ -109,8 +124,17 @@ instance Arbitrary UserInput where
   arbitrary = UI . C8.pack <$> arbitrary
 
 --------------------------------------------------------------------------------
+cipherInitNoError :: Cipher c => c -> ByteString -> c
+cipherInitNoError _ k = case cipherInit k of
+  CryptoPassed a -> a
+  CryptoFailed e -> error (show e)
+
+--------------------------------------------------------------------------------
 newRNCryptorContext :: ByteString -> RNCryptorHeader -> RNCryptorContext
 newRNCryptorContext userKey hdr =
-  let eKey = sha1PBKDF2 userKey (rncEncryptionSalt hdr) 10000 32
-      cipher = initAES eKey
-  in RNCryptorContext hdr cipher
+  let hmacSalt = rncHMACSalt hdr
+      hmacKey  = makeKey userKey hmacSalt
+      hmacCtx  = initialize hmacKey::Context SHA256
+      encKey   = makeKey userKey $ rncEncryptionSalt hdr
+      cipher   = cipherInitNoError (undefined::AES128) encKey
```

Comment:
See comment about undefined above.  Can't take this out without hitting a compiler error.